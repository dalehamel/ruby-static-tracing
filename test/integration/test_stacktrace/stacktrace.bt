usdt::stacktrace_operation:call
{
  printf("%s", str(arg1))
}
