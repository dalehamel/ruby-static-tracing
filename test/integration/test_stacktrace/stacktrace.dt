stacktrace_operation*:::call
{
  printf("Attaching 1 probe...\n%s\n\n", copyinstr(arg1))
}
