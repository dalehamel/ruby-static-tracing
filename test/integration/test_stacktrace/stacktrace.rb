require 'ruby-static-tracing'
STDOUT.sync = true

class StacktraceOperation
  def execute
    1 + 1
  end

  def call
    execute
  end

  StaticTracing::Tracer::Stack.register(self, :call)
end

StaticTracing.configure do |config|
  config.add_tracer(StaticTracing::Tracer::Stack)
end

stack_operation = StacktraceOperation.new

Signal.trap('USR2') do
  stack_operation.call
end

while true
end
