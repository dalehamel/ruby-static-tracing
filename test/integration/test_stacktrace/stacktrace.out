Attaching 1 probe...
stacktrace.rb:23:in `block in <main>'
stacktrace.rb:26:in `<main>'

